api/v1alpha1/etcdpeer_types.go

```diff
@@ -7,10 +7,26 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// BootstrapPeer describes a single *other* peer that will be used for bootstrapping
+type BootstrapPeer struct {
+	// Name is the initial name of the peer, which is set into etcd at the
+	// end of this process
+	Name string `json:"name"`
+
+	// Host forms part of the Advertise URL - the URL at which this peer can
+	// be contacted. The port and scheme are hardcoded.
+	Host string `json:"host"`
+}
+
 // EtcdPeerSpec defines the desired state of EtcdPeer
 type EtcdPeerSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
+
+	// BootstrapPeers is the list of all other peers for bootstrap purpses.
+	// This does not need to be kept up to date after bootstrap has occured
+	// as it's *only* used before cluster peers have conatacted each other.
+	BootstrapPeers []BootstrapPeer `json:"bootstrapPeers,omitempty"`
```

Comment:
Can you refactor this to have a structure like here? https://github.com/munnerz/etcd-operator-poc/blob/master/api/v1alpha1/etcdpeer_types.go#L35-L54

```
        ...

	// Bootstrap defines enough bootstrap configuration for this peer to join
	// an existing cluster or form a new one if needed.
	// This configuration is only used when a new cluster is created or when a
	// new peer is being introduced.
	Bootstrap *BootstrapConfig `json:"bootstrap"`
}

// BootstrapConfig defines how an etcd cluster should be bootstrapped.
type BootstrapConfig struct {
	// Static contains bootstrap configuration that can be used to bootstrap
	// an etcd cluster using 'static' peer configuration.
	Static *StaticBootstrapConfig `json:"static"`
}

// StaticBootstrapConfig is used to configure the 'static' bootstrap method
type StaticBootstrapConfig struct {
	// The initial list of peers in the cluster used when bootstrapping a peer.
	InitialClusterPeers []PeerAddress `json:"initialClusterPeers,omitempty"`
}
```

I think I mentioned it on the last PR, but even though we are unlikely to support more kinds of bootstrap configs in the short-term, we definitely want to leave room for it in our API 😄

Reply:
Yeah works for me :+1: